app/pages/lab/workflow-components/tutorials.jsx

```diff
@@ -3,31 +3,37 @@ import apiClient from 'panoptes-client/lib/api-client';
 
 export default function Tutorials({ project, workflow }) {
   const [tutorials, setTutorials] = useState([]);
-  const [workflowTutorial, setWorkflowTutorial] = useState(null);
+  const [linkedTutorials, setLinkedTutorials] = useState([]);
 
   useEffect(function loadTutorials() {
-    Promise.all([
-      apiClient.type('tutorials')
-        .get({ project_id: project.id, page_size: 100 })
-        .catch(() => []),
-      apiClient.type('tutorials')
-        .get({ workflow_id: workflow.id, page_size: 100 })
-        .catch(() => [])
-    ])
-    .then(([projectTutorials, workflowTutorials]) => {
-      const tutorials = projectTutorials.filter(value => value.kind === 'tutorial' || value.kind === null);
-      const [workflowTutorial] = tutorials.filter(value => workflowTutorials.includes(value));
-      setTutorials(tutorials);
-      setWorkflowTutorial(workflowTutorial);
-    });
+    apiClient.type('tutorials')
+      .get({ project_id: project.id, page_size: 100 })
+      .catch(() => [])
+      .then(tutorials => {
+        setTutorials(tutorials.filter(value => value.kind === 'tutorial' || value.kind === null));
+      });
+    apiClient.type('tutorials')
+      .get({ workflow_id: workflow.id, page_size: 100 })
+      .catch(() => [])
+      .then(tutorials => {
+        setLinkedTutorials(tutorials.filter(value => value.kind === 'tutorial' || value.kind === null));
+      });
   }, [project?.id, workflow?.id]);
 
-  function removeTutorial() {
-    setWorkflowTutorial(null)
-    return workflow.removeLink('tutorials', workflowTutorial?.id);
+  function updateTutorialLinks(linkedTutorial) {
+    linkedTutorials
+      .filter(tutorial => tutorial.id !== linkedTutorial?.id)
+      .forEach(tutorial => workflow.removeLink('tutorials', tutorial?.id));
+    if (linkedTutorial) {
+      workflow.addLink('tutorials', [linkedTutorial.id]);
+      setLinkedTutorials([linkedTutorial]);
+    } else {
+      setLinkedTutorials([]);
+    }
   }
 
   function onChange(event) {
+    // TODO: radio groups only fire change events for the checked input, so what does this line do?
```

Comment:
Is this comment intentionally left in the PR?

To avoid future devs from "what's this comment for?" I recommend addressing the TODO in this PR or opening a separate Issue for it.

Reply:
This is intentional. What is that line doing? I don’t understand it.

Reply:
Is it safe to remove the following line?

Reply:
I've removed `shouldAdd` but I wish I knew why it was there in the first place. 🤔

Reply:
The workflow editor still works without it. 🤷🏻‍♂️